ports/raspberrypi/common-hal/microcontroller/Processor.c

```diff
@@ -59,6 +61,16 @@ uint32_t common_hal_mcu_processor_get_frequency(void) {
     return clock_get_hz(clk_sys);
 }
 
+uint32_t common_hal_mcu_processor_set_frequency(mcu_processor_obj_t *self,
+    uint32_t frequency) {
+    uint32_t freq = frequency / 1000;
+
+    if (!set_sys_clock_khz(freq, false)) {
+        mp_raise_ValueError(translate("Invalid frequency supplied"));
+    }
+    return clock_get_hz(clk_sys);
+}
```

Comment:
```suggestion
void common_hal_mcu_processor_set_frequency(mcu_processor_obj_t *self, uint32_t frequency) {
    if (!set_sys_clock_khz(frequency / 1000, false)) {
        mp_arg_error_invalid(MP_QSTR_frequency);
    }
}
```
Not related to your changes but if you are at it, can you please change the signature of the function:
```patch
- uint32_t common_hal_mcu_processor_set_frequency
+ void common_hal_mcu_processor_set_frequency
```
The changes need to be made at the following 3 places:
https://github.com/adafruit/circuitpython/blob/a84a8855ed2deff2b20ef5cfd11bbdef862fc239/ports/mimxrt10xx/common-hal/microcontroller/Processor.c#L55
https://github.com/adafruit/circuitpython/blob/a84a8855ed2deff2b20ef5cfd11bbdef862fc239/ports/raspberrypi/common-hal/microcontroller/Processor.c#L64
https://github.com/adafruit/circuitpython/blob/a84a8855ed2deff2b20ef5cfd11bbdef862fc239/shared-bindings/microcontroller/Processor.h#L42